tests/private-active-active/proxy.tf

```diff
@@ -62,3 +63,34 @@ resource "aws_security_group_rule" "proxy_egress" {
 
   security_group_id = aws_security_group.proxy.id
 }
+
+resource "aws_iam_instance_profile" "proxy_ssm" {
+  name_prefix = "${local.friendly_name_prefix}-proxy-ssm"
+  role        = aws_iam_role.proxy_instance_role.name
+}
+
+resource "aws_iam_role" "proxy_instance_role" {
+  name_prefix        = "${local.friendly_name_prefix}-proxy-ssm"
+  assume_role_policy = data.aws_iam_policy_document.proxy_instance_role.json
+
+  tags = local.common_tags
+}
+
+data "aws_iam_policy_document" "proxy_instance_role" {
+  statement {
+    effect = "Allow"
+    actions = [
+      "sts:AssumeRole",
+    ]
+
+    principals {
+      type        = "Service"
+      identifiers = ["ec2.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_role_policy_attachment" "ssm" {
+  role       = aws_iam_role.proxy_instance_role.name
+  policy_arn = "arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore"
```

Comment:
Since this ARN string appears more than once in this test module, I think we should store it as a local value. 📌